### PR TITLE
#163387853 Initial server setup

### PR DESCRIPTION
**Pull request by Raymond-Osy**
#### What does this PR do?
This is the initial server setup for this application

#### Description of Task to be completed?
- Add a package.json file
- Add the express app
- Add babel for transpiling Es6 syntax down
- Add Eslint and use the airbnb style guide for linting
- Add the .gitignore file for git to ignore large files

#### How should this be manually tested?
- Do a git clone in your local computer
- Open the console/comandline
- type npm run dev
- Open a browser and visit http://localhost:3000/ then you should see a response to show that the server is up and running

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#163387853](https://www.pivotaltracker.com/n/projects/2239146)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({
+  extended: true,
+}));
+
+
+app.all('*', (req, res) => {
+  res.status(404).send('The page you are looking for was not found');
```
Comment (yomigeek) [inline server/app.js:14]: Good work @Raymond-Osy However, Please check the project doc for more clarity on the format of the response message.

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+});
+
+const port = parseInt(process.env.PORT, 10) || 8000;
+app.set('port', port);
+
+const server = http.createServer(app);
```
Comment (yomigeek) [inline server/app.js]: @Raymond-Osy Please, can you explain why this is needed?

**Review by yomigeek — CHANGES_REQUESTED** [review]

File: server/app.js (hunk 1 above)
Comment (Raymond-Osy) [reply server/app.js:14]: Alright, Noted

File: server/app.js (hunk 2 above)
Comment (Raymond-Osy) [reply server/app.js]: On line 19, the port is set to an environment variable, if none is found, it uses the specified port, in my case port 8000. On line 22, i simply created the server using the express application.

Comment (yomigeek) [reply server/app.js]: @Raymond-Osy To use express and http.createServer in the same application, as they fit the same purpose but express is a more useful abstraction over http.
In express, listen boots the app. Before booting, you need to setup handlers and middlewares for routes (i.e. get, post, use) which you have already done. So in this case using listen is enough.

Comment (Raymond-Osy) [reply server/app.js]: Alright noted. I shall act on this feedback....Thanks

Comment (Raymond-Osy) [reply server/app.js]: Thanks for the feedback. I have acted on it. Can i go ahead and merge?

**Pull request merged**